platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py

```diff
@@ -792,6 +793,21 @@ def get_xcvr_api(self):
                 self._xcvr_api.get_tx_fault = self.get_tx_fault
         return self._xcvr_api
 
+    def is_sw_control(self):
+        if not DeviceDataManager.is_independent_mode():
+            return False
+
+        db = utils.DbUtils.get_db_instance('STATE_DB')
+        control_type = db.get('STATE_DB', f'TRANSCEIVER_MODULES_MGMT|{self.sdk_index}', 'control_type')
```

Comment:
@Junchao-Mellanox as discussed, we decided to use existing TRANSCEIVER_INFO table to distinguish CMIS (SW_CONTROL) vs NON-CMIS (FW_CONTROL)?